src/Features/CSharp/Portable/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.Rewriting.cs

```diff
@@ -0,0 +1,133 @@
+                    ? null
+                    : AsWhenClause(section.Labels[0]),
+                expression: AsExpressionSyntax(section.Body));
+        }
+
+        private static ExpressionSyntax AsExpressionSyntax(IOperation operation)
+        {
+            return operation switch
+            {
+                IReturnOperation op => (ExpressionSyntax)op.ReturnedValue.Syntax,
+                IThrowOperation op => ThrowExpression((ExpressionSyntax)op.Exception.Syntax),
+                IBlockOperation op => AsExpressionSyntax(op.Operations.Single()),
+                var v => throw ExceptionUtilities.UnexpectedValue(v.Kind)
+            };
+        }
+
+        public override SyntaxNode CreateSwitchStatement(IfStatementSyntax ifStatement, SyntaxNode expression, IEnumerable<SyntaxNode> sectionList)
+        {
+            var block = ifStatement.Statement as BlockSyntax;
+            return SwitchStatement(
+                switchKeyword: Token(SyntaxKind.SwitchKeyword).WithTriviaFrom(ifStatement.IfKeyword),
+                openParenToken: ifStatement.OpenParenToken,
+                expression: (ExpressionSyntax)expression,
+                closeParenToken: ifStatement.CloseParenToken.WithPrependedLeadingTrivia(ElasticMarker),
+                openBraceToken: block?.OpenBraceToken ?? Token(SyntaxKind.OpenBraceToken),
+                sections: List(sectionList.Cast<SwitchSectionSyntax>()),
+                closeBraceToken: block?.CloseBraceToken ?? Token(SyntaxKind.CloseBraceToken));
+        }
+
+        private static WhenClauseSyntax? AsWhenClause(AnalyzedSwitchLabel label)
+            => AsWhenClause(label.Guards
+                .Select(e => e.WalkUpParentheses())
+                .AggregateOrDefault((prev, current) => BinaryExpression(SyntaxKind.LogicalAndExpression, current, prev)));
+
+
+        private static WhenClauseSyntax? AsWhenClause(ExpressionSyntax? expression)
+            => expression is null ? null : WhenClause(expression);
+
+        public override SyntaxNode AsSwitchLabelSyntax(AnalyzedSwitchLabel label)
+            => CasePatternSwitchLabel(
+                AsPatternSyntax(label.Pattern),
+                AsWhenClause(label),
+                Token(SyntaxKind.ColonToken));
+
+        private static PatternSyntax AsPatternSyntax(AnalyzedPattern pattern)
+            => pattern switch
+            {
+                AnalyzedPattern.Constant p => ConstantPattern(p.ExpressionSyntax),
+                AnalyzedPattern.Source p => p.PatternSyntax,
+                AnalyzedPattern.Type p => DeclarationPattern((TypeSyntax)p.IsExpressionSyntax.Right, DiscardDesignation()),
+                var p => throw ExceptionUtilities.UnexpectedValue(p)
+            };
+
+        public override IEnumerable<SyntaxNode> AsSwitchSectionStatements(IOperation operation)
+        {
+            var node = operation.Syntax;
+            var requiresBreak = operation.SemanticModel.AnalyzeControlFlow(node).EndPointIsReachable;
```

Comment:
consider using HasUnreachableEndPoint

Reply:
It's defined in the inner Analyzer class.